devito/symbolics/printer.py

```diff
@@ -94,6 +96,23 @@ def _print_Mod(self, expr):
         result = '%'.join(args)
         return result
 
+    def _print_Abs(self, expr):
+        """Print an absolute value. Use `abs` if can infer it is an Integer"""
+        # `abs` undefined in soem offloading cases
+        if isinstance(configuration['platform'], Device):
```

Comment:
we can't access configuration in a printer... it escapes all abstractions (e.g., this would break if platform=cpu but one creates `Operator(..., platform=nvidiaX)`

nvidia definitely supports abs/fabs/... see for example here: https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__SINGLE.html#group__CUDA__MATH__SINGLE_1gb00f8593e1bfb1985526020fbec4e0fc

where did you see the failure? perhaps it's just missing an include? though math.h should suffice...

Reply:
Is an AMD issue with aomp looks like

https://github.com/devitocodes/devito/actions/runs/3242390619/jobs/5315585337

Reply:
Let me take a look at his